File: src/main/java/fr/epyi/metropiamod/capabilities/SkinData.java
package fr.epyi.metropiamod.capabilities;

import java.util.ArrayList;

public class SkinData implements ISkinData {

    private ArrayList<String> urls = new ArrayList<String>(0);
    private String bodyType = "default";

    @Override
    public ArrayList<String> getSkin() {
        return this.urls;
    }

    @Override
    public void setSkin(ArrayList<String> url) {
        this.urls = url == null ? new ArrayList<String>() : url;
    }

    @Override
    public String getModelType() {
        return this.bodyType;
    }

    @Override
    public void setModelType(String bodyType) {
        this.bodyType = bodyType;
    }
}

File: src/main/java/fr/epyi/metropiamod/capabilities/SkinData.kt
package fr.epyi.metropiamod.capabilities

class SkinData : ISkinData {

    private var urls = ArrayList<String?>()
    private var bodyType: String = "default"

    override fun getSkin(): ArrayList<String?> {
        return this.urls
    }

    override fun setSkin(urls: ArrayList<String?>) {
        this.urls = urls
    }

    override fun getModelType(): String {
        return this.bodyType
    }

    override fun setModelType(url: String) {
        this.bodyType = url
    }
}


